rpc/websockets.go

```diff
@@ -186,100 +186,106 @@ func (s *websocketsServer) readLoop(wsConn *wsConn) {
 			return
 		}
 
-		var msg map[string]interface{}
-		if err = json.Unmarshal(mb, &msg); err != nil {
-			s.sendErrResponse(wsConn, err.Error())
-			continue
-		}
-
-		// check if method == eth_subscribe or eth_unsubscribe
-		method, ok := msg["method"].(string)
-		if !ok {
-			// otherwise, call the usual rpc server to respond
+		if isBatch(mb) {
 			if err := s.tcpGetAndSendResponse(wsConn, mb); err != nil {
 				s.sendErrResponse(wsConn, err.Error())
 			}
-
-			continue
-		}
-
-		connID, ok := msg["id"].(float64)
-		if !ok {
-			s.sendErrResponse(
-				wsConn,
-				fmt.Errorf("invalid type for connection ID: %T", msg["id"]).Error(),
-			)
-			continue
-		}
-
-		switch method {
-		case "eth_subscribe":
-			params, ok := msg["params"].([]interface{})
-			if !ok {
-				s.sendErrResponse(wsConn, "invalid parameters")
-				continue
-			}
-
-			if len(params) == 0 {
-				s.sendErrResponse(wsConn, "empty parameters")
-				continue
-			}
-
-			subID := rpc.NewID()
-			unsubFn, err := s.api.subscribe(wsConn, subID, params)
-			if err != nil {
+		} else {
```

Comment:
nit, use `continue` in the `if` above and drop the `else` here

Reply:
thanks, much cleaner